main.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/cyverse-de/configurate"
+	l "github.com/cyverse-de/go-mod/logging"
+	"github.com/cyverse-de/go-mod/otelutils"
+	"github.com/cyverse-de/messaging/v9"
+
+	"github.com/cyverse-de/group-propagator/client/datainfo"
+	"github.com/cyverse-de/group-propagator/client/groups"
+	"github.com/cyverse-de/group-propagator/logging"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"github.com/streadway/amqp"
+)
+
+var log = logging.Log.WithFields(logrus.Fields{"package": "main"})
+
+const serviceName = "group-propagator"
+
+const otelName = "github.com/cyverse-de/group-propagator"
+
+func getQueueName(prefix string) string {
+	if len(prefix) > 0 {
+		return fmt.Sprintf("%s.%s", prefix, serviceName)
+	}
+	return serviceName
+}
+
+// A spinner to keep the program running since client.Listen() needs to be in a goroutine.
+//nolint
+func spin() {
+	spinner := make(chan int)
+	for {
+		select {
+		case <-spinner:
+			fmt.Println("Exiting")
+			break
+		}
+	}
+}
+
+func main() {
+	var (
+		cfgPath  = flag.String("config", "/etc/iplant/de/group-propagator.yml", "The path to the config file")
+		logLevel = flag.String("log-level", "info", "One of trace, debug, info, warn, error, fatal, or panic.")
+
+		err error
+		cfg *viper.Viper
+	)
+
+	flag.Parse()
+	l.SetupLogging(*logLevel)
+
+	var tracerCtx, cancel = context.WithCancel(context.Background())
+	defer cancel()
+	shutdown := otelutils.TracerProviderFromEnv(tracerCtx, serviceName, func(e error) { log.Fatal(e) })
+	defer shutdown()
+
+	if *cfgPath == "" {
+		log.Fatal("--config must not be the empty string")
+	}
+
+	if cfg, err = configurate.Init(*cfgPath); err != nil {
+		log.Fatal(err.Error())
+	}
+
+	// package up config nicely
+
+	// Set up AMQP
+	listenClient, err := messaging.NewClient(cfg.GetString("amqp.uri"), true)
```

Comment:
Another thing I should do in this PR is make a proper configuration object that validates things, and pass that around instead of using a bunch of `cfg.GetString` type calls. (That's on the list, basically)